src/selectors/entities/teams.js

```diff
@@ -124,3 +133,84 @@ export const getJoinableTeams = createSelector(
         return openTeams;
     }
 );
+
+export const getJoinableTeamIds = createIdsSelector(
+    getTeams,
+    getTeamMemberships,
+    (teams, myMembers) => {
+        return Object.keys(teams).filter((id) => {
+            const team = teams[id];
+            const member = myMembers[id];
+            return team.allow_open_invite && !member;
+        });
+    }
+);
+
+export const getMySortedTeamIds = createIdsSelector(
+    getTeams,
+    getTeamMemberships,
+    (state, locale) => locale,
+    (teams, myMembers, locale) => {
+        return Object.values(teams).filter((t) => myMembers[t.id]).sort((a, b) => {
+            if (a.display_name !== b.display_name) {
+                return a.display_name.toLowerCase().localeCompare(b.display_name.toLowerCase(), locale, {numeric: true});
+            }
+
+            return a.name.toLowerCase().localeCompare(b.name.toLowerCase(), locale, {numeric: true});
+        }).map((t) => t.id);
+    }
+);
+
+export const getMyTeamsCount = createSelector(
+    getTeamMemberships,
+    (teams) => {
+        return Object.keys(teams).length;
+    }
+);
+
+// Returns the number of mentions or -1 if it has unreads
+// for every team except the current one
+export const getTeamsMentions = createSelector(
+    getCurrentTeamId,
+    getTeamMemberships,
+    (currentTeamId, teamMembers) => {
+        let mentionCount = 0;
+        let messageCount = 0;
+        Object.values(teamMembers).forEach((m) => {
+            if (m.team_id !== currentTeamId) {
+                mentionCount = mentionCount + (m.mention_count || 0);
+                messageCount = messageCount + (m.msg_count || 0);
+            }
+        });
+
+        let badgeCount = 0;
+        if (mentionCount) {
+            badgeCount = mentionCount;
+        } else if (messageCount) {
+            badgeCount = -1;
+        }
+
+        return badgeCount;
+    }
+);
+
+// Returns the number of mentions or -1 if it has unreads
+// for every team except the current one
+export const getTeamMentions = createSelector(
```

Comment:
This should be done as a selector factory (ie `makeGetTeamMentions`) or else the memoization does nothing since it'll constantly be called with different ids

Reply:
Similarly, the name is a bit unclear. Maybe `makeGetBadgeCountForTeamId` to make its use clear and point to the fact that you need to pass in a team id?

Reply:
I'll create the factory and update the name here